watershed_workflow/land_cover_properties_Soumendra.py

```diff
@@ -0,0 +1,328 @@
+    """Compute a map from NLCD indices to MODIS indices using correlation
+    of the two rasters.
+
+    Parameters
+    ----------
+    modis_lc : np.ndarray[NX,NY]
+      Raster of a single year of MODIS land cover type.
+    modis_profile : dict
+      Rasterio profile of the modis_lc
+    nlcd_lc : np.ndarray[NX,NY]
+      Raster of a single year of NLCD land cover type.
+    nlcd_profile : dict
+      Rasterio profile of the nlcd_lc
+    warp : bool, optional=True
+      Warp MODIS into NLCD raster profile.
+
+    Returns
+    -------
+    crosswalk
+       Maps keys of MODIS indices into NLCD indices.
+
+    """
+    if warp:
+        assert(watershed_workflow.crs.equal(modis_profile['crs'], nlcd_profile['crs']))
+        modis_profile, modis_lc = watershed_workflow.warp.raster(modis_profile, modis_lc,
+                                                             nlcd_profile['crs'],
+                                                             nlcd_profile['resolution'],
+                                                             nlcd_profile['height'],
+                                                             nlcd_profile['width'])
+
+    unique_nlcd = [val for val in np.unique(nlcd_lc.ravel()) if val != nlcd_lc.fill_value]
+    unique_modis = [val for val in np.unique(modis_lc.ravel()) if val != modis_lc.fill_value]
+
+
+    #correlation_matrix = np.array((len(unique_nlcd), len(unique_modis)), 'd')
+    correlation_matrix = np.zeros((len(unique_nlcd), len(unique_modis)), dtype=float, order='C')
+    
+
+    for i, nlcd in enumerate(unique_nlcd):
+        where = np.where(nlcd_lc == nlcd)
+        count_nlcd = len(where[0])
+
+        for j, modis in enumerate(unique_modis):
+            where_modis = np.where(np.bitwise_and(nlcd_lc == nlcd, modis_lc == modis))
+            count_modis_and_nlcd = len(where_modis[0])
+    
+            correlation_matrix[i,j] =  count_modis_and_nlcd / count_nlcd
+
+    if plot:
+        plt.imshow(correlation_matrix)
+        plt.show()
+
+    crosswalk = dict()
+    for i, nlcd in enumerate(unique_nlcd):
+        crosswalk[nlcd] = unique_modis[np.argmax(correlation_matrix[i])]
+    return crosswalk
+                                  
+def write_to_ats(lai_raster, crosswalk, smooth=False, smooth_filter=False, nyears=None):
+    """Accepts a numpy array of LAI data and returns a dictionary ATS data.
+    
+    Parameters:
+        lai_raster : dict
+            MODIS LAI dictinary
+        smooth : bool
+            whether to smooth the data or not
+        smooth_filter : bool
+            whether to apply a savgol filter for smoothing. Default is False.
+        nyears : int
+            how many years to repeat the typical year. Default is to get the same length as the raw data.
+        crosswalk : Maps keys of MODIS indices into NLCD indices.
+    """
+
+    outputs['modis_filename'] = f'../data-processed/{name}_MODIS_LAI_smoothed_2002_2020.h5'
```

Comment:
This can't run, outputs is not defined.

Probably this function needs to accept as an argument a filename?